openshift/machine-config-operator
# MCO-664: adds on-cluster build dev preview docs

**- What I did**

Adds on-cluster build docs. There are two docs and a diagram as described below:
- `MachineOSBuildDesign.md` - This doc is primarily for MCO developers to understand how the on-cluster builds mechanism works. It includes a description of the various components such as BuildController, ImageBuilder, and CustomPodBuilder as well as an SVG version of my diagram outlining these things.
- `OnClusterBuildInstructions.md` - This doc is primarily for cluster admins wishing to give on-cluster builds a try in their dev preview clusters.
- `MachineOSBuilderDiagram.svg` - This is the aforementioned diagram in SVG format.

**- How to verify it**

Run through the instructions and see if you can bring up a 4.14 cluster with a custom layered OS.

**- Description for the changelog**
Adds on-cluster build dev preview docs
